system/jlib/jstring.cpp

```diff
@@ -505,30 +505,30 @@ void StringBuffer::setLength(unsigned len)
 
 size32_t StringBuffer::lengthUtf8() const
 {
-    size32_t chars = 0;
-    for (unsigned offset=0; offset < curLen; offset += readUtf8Size(buffer+offset))
+    size_t chars = 0;
+    for (size_t offset=0; offset < curLen; offset += readUtf8Size(buffer+offset))
         chars++;
     return chars;
```

Comment:
Should explicitly cast to (size32_t) - to avoid warnings on some compilers. With a comment?

Reply:
but I guess really the utf methods should return size_t too...

Reply:
'chars' can be a size32_t

Reply:
Am going to change/add cast + comment.